src/module.c

```diff
@@ -9857,40 +9861,67 @@ void moduleFreeModuleStructure(struct RedisModule *module) {
     zfree(module);
 }
 
+/* Free a parent_cmd: freeCommand(parent_cmd, NULL)
+ * Free a sub_command: freeCommand(sub_cmd, parent_cmd) */
+void freeCommand(struct redisCommand *cmd, struct redisCommand *parent_cmd) {
```

Comment:
1. maybe rename to `moduleFreeCommand`
2. i think a nicer API would be `moduleFreeCommand(struct redisCommand *cmd) where it looks song like:

```
void moduleFreeCommand(struct redisCommand *cmd) {
    // free stuff
    if (subcommands) {
        // for command in subcommand, call moduleFreeCommand + delete from subcommands_dict
    } else {
        dictDelete(server.commands, cmd->fullname);
        dictDelete(server.orig_commands, cmd->fullname);
    }
}
```